File: oep_spring_backend/src/main/java/com/oep/entities/Batch.java
package com.oep.entities;

import java.time.LocalDate;
import jakarta.persistence.AttributeOverride;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Index;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;


@Entity
@AttributeOverride(name = "id", column = @Column(name = "batch_id"))
@Table(name = "batches", indexes = {
		@Index(name = "idx_batch_batch_name", columnList = "batch_name"),
		@Index(name = "idx_batch_start_date", columnList = "start_date"),
		@Index(name = "idx_batch_end_date", columnList = "end_date")
})
@Getter
@Setter
public class Batch extends BaseEntity{
	@Column(name = "batch_name", length = 30, nullable = false, unique = true)
	private String batchName;
	@Column(name = "start_date", nullable = false)
	private LocalDate startDate;
	@Column(name = "end_date", nullable = false)
	private LocalDate endDate;
}

File: AdminServiceDotNET/Models/Batch.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace AdminServiceDotNET.Models
{
    [Table("batches")]
    [Index(nameof(BatchName), Name = "idx_batch_batch_name")]
    [Index(nameof(StartDate), Name = "idx_batch_start_date")]
    [Index(nameof(EndDate), Name = "idx_batch_end_date")]
    public class Batch
    {
        [Key]
        [Required]
        [Column("batch_id")]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; }

        [Required]
        [Column("batch_name")]
        [MaxLength(30)]
        public string BatchName { get; set; }

        [Column("description")]
        [MaxLength(500)]
        public string? Description { get; set; }

        [Required]
        [Column("start_date")]
        public DateTime StartDate { get; set; }

        [Required]
        [Column("end_date")]
        public DateTime EndDate { get; set; }

        [Required]
        [Column("is_active")]
        public bool IsActive { get; set; } = true;
    }
}

